### PR TITLE
External Operators 2

### PR DESCRIPTION
**Pull request by samskalicky**
## Description ##
This is a follow-on PR after the initial investigation in #18904 to support loading libraries containing MXNet backend components (particularly operators). Since #19241, the version script was removed and now Linux builds of `libmxnet.so` expose all MXNet symbols. This opens the door for dynamically loading libraries that depend on `libmxnet.so`.

This PR adds a build option `BUILD_EXTENSION` to the CMakeLists.txt file to give a path to the extension directory to build. This enables the existing MXNet build infrastructure to provide all includes directories automatically which simplifies building extensions. 

An example is also added in `examples/extensions/lib_external_ops` that builds an operator that is equivalent to a built-in/backend operator into a shared object and uses the existing `mx.library.load` API to load that library at runtime. 

### Caveots
Windows is not supported since by default all symbols are hidden in `mxnet.dll` so dynamically loading libraries will not work. Similarly with Mac, the version script does not expose the necessary symbols at this time. These other platforms can be considered in the future based on user request & community motivation.

### CODE REVIEW

File: example/extensions/lib_external_ops/test_loading.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# pylint: disable=arguments-differ
+
+# This test checks if dynamic loading of library into MXNet is successful
+# and checks the computation of an external operator
+
+import mxnet as mx
+import os
+
+# check if operator exists
+if hasattr(mx.nd,'min_ex'):
```
Comment (wkcn) [inline example/extensions/lib_external_ops/test_loading.py:30]: Thank you!
Could you please format the code in the line 30th, 36th, and 38th ?

**Review by wkcn — COMMENTED** [review]
@samskalicky Thank you!
I have two suggestions : )

File: example/extensions/lib_external_ops/test_loading.py (hunk 1 above)
Comment (samskalicky) [reply example/extensions/lib_external_ops/test_loading.py]: thanks for point this out, fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -89,6 +89,7 @@ option(USE_TENSORRT "Enable inference optimization with TensorRT." OFF)
 option(USE_ASAN "Enable Clang/GCC ASAN sanitizers." OFF)
 cmake_dependent_option(ENABLE_TESTCOVERAGE "Enable compilation with test coverage metric output" OFF "NOT MSVC" OFF)
 option(USE_INT64_TENSOR_SIZE "Use int64_t to represent the total number of elements in a tensor" OFF)
+option(BUILD_EXTENSION "Path to extension to build" "")
```
[the remark this answers is not in the document]
Comment (samskalicky) [reply CMakeLists.txt]: Great idea! I like the `BUILD_EXTENSION_PATH` so I changed it to that, thanks!

**Review by wkcn — APPROVED** [review]
LGTM!
Thank you for the contribution!

**Review by mseth10 — APPROVED** [review]
LGTM

**Comment by samskalicky** [comment]
Thanks @mseth10 for adding the tests to the CI! 

@leezu anything else we need here before taking the next step?

**Review by leezu — APPROVED** [review]

**Pull request merged**